aten/src/ATen/core/ivalue.cpp

```diff
@@ -304,6 +307,17 @@ std::vector<std::pair<IValue, IValue>> iterationOrder(const c10::Dict<IValue, IV
   return ordered;
 }
 
+StrongTypePtr::StrongTypePtr(
+    std::shared_ptr<torch::jit::script::CompilationUnit> cu,
+    std::shared_ptr<Type> type) {
+  cu_ = std::move(cu);
+  type_ = type;
+  TORCH_INTERNAL_ASSERT(type_);
+  if (type_->cast<ClassType>()) {
+    TORCH_INTERNAL_ASSERT(cu_);
```

Comment:
Can we add a more detailed error message in this case? Also, one useful check might be to assert that the class type's owning compilation unit is the same as the `cu` we passed in here.

Reply:
will do